library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -2220,6 +2220,39 @@ void libspdm_reset_context(void *context)
     #endif /* LIBSPDM_ENABLE_MSG_LOG */
 }
 
+/**
+ * Free the SPDM context: if there are some memory need free, the function will free them.
+ *
+ * @param[in]  spdm_context         A pointer to the SPDM context.
+ * @param[in]  is_requester        Indicate of the key generation for a requester or a responder.
+ *
+ */
+void libspdm_deinit_context(void *context, bool is_requester)
+{
+#if !LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
```

Comment:
Even if `LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT` is enabled this would still be needed right?

Reply:
When `LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT` is enabled, the `spdm_context->connection_info.peer_used_cert_chain[slot_id].leaf_cert_public_key` will be not exist. But I think we can make the `LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT` in the `libspdm_deinit_context`, because integrator can use the function easily: Just call `libspdm_deinit_context`, no matter weather `LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT` is existed.